fsspec/implementations/reference.py

```diff
@@ -812,14 +812,16 @@ def cat(self, path, recursive=False, on_error="raise", **kwargs):
                         raise
                     if on_error != "omit":
                         out[p] = err
+                else:
+                    valid_paths.append(p)
```

Comment:
Perhaps it would be clearer if this were inside the `try` block, to show that there must be the same number of urls, starts, ends, valid_paths when we are done. Alternatively, all four come here in the `else` block, since it's only the `_cat_common` line which can error.

Reply:
that makes sense, I will adjust it. All under `else:` seems the cleanest to me